org.knime.scijava.commands/src/org/knime/scijava/commands/settings/DefaultSettingsModelTypeService.java

```diff
@@ -20,60 +25,85 @@
 @SuppressWarnings("rawtypes")
 @Plugin(type = SettingsModelTypeService.class)
 public class DefaultSettingsModelTypeService
-		extends AbstractSingletonService<SettingsModelTypePlugin>
-		implements SettingsModelTypeService {
-
-	private final Map<Class<? extends SettingsModel>, SettingsModelTypePlugin> m_pluginsByModel = new HashMap<>();
-	private final Map<Class<?>, SettingsModelTypePlugin> m_pluginsByValue = new HashMap<>();
-
-	@Override
-	public Class<SettingsModelTypePlugin> getPluginType() {
-		return SettingsModelTypePlugin.class;
-	}
-
-	@Override
-	public SettingsModelType getSettingsModelTypeFor(
-			final SettingsModel settingsModel) {
-
-		final SettingsModelTypePlugin plugin = m_pluginsByModel
-				.get(settingsModel.getClass());
-		if (plugin != null) {
-			return plugin;
-		}
-
-		for (final SettingsModelTypePlugin p : getInstances()) {
-			if (p.getSettingsModelClass().isInstance(settingsModel)) {
-				m_pluginsByModel.put(settingsModel.getClass(), p);
-				return p;
-			}
-		}
-		// nothing found
-		return null;
-	}
-
-	@Override
-	public SettingsModelType getSettingsModelTypeFor(final Class<?> value) {
-
-		// check cache
-		final SettingsModelTypePlugin plugin = m_pluginsByValue.get(value);
-		if (plugin != null) {
-			return plugin;
-		}
-
-		// check primitive conversion cache
-		final Class<?> checkValue = PrimitiveTypeUtils
-				.convertIfPrimitive(value);
-
-		// search
-		for (final SettingsModelTypePlugin<?, ?> p : getInstances()) {
-			if (p.getValueClass().isAssignableFrom(checkValue)) {
-				m_pluginsByValue.put(value, p);
-				return p;
-			}
-		}
-
-		// nothing found
-		return null;
-	}
+        extends AbstractSingletonService<SettingsModelType>
+        implements SettingsModelTypeService {
+
+    @Parameter
+    private ConverterCacheService cs;
+
+    private final Map<Class<? extends SettingsModel>, SettingsModelType> m_pluginsByModel = new HashMap<>();
+    private final Map<Class<?>, SettingsModelType> m_pluginsByValue = new HashMap<>();
+
+    @Override
+    public SettingsModelType getSettingsModelTypeFor(
+            final SettingsModel settingsModel) {
+
+        final SettingsModelType plugin = m_pluginsByModel
+                .get(settingsModel.getClass());
+        if (plugin != null) {
+            return plugin;
+        }
+
+        for (final SettingsModelType p : getInstances()) {
+            if (p.getSettingsModelClass().isInstance(settingsModel)) {
+                m_pluginsByModel.put(settingsModel.getClass(), p);
+                return p;
+            }
+        }
+        // nothing found
+        return null;
+    }
+
+    @Override
+    public SettingsModelType getSettingsModelTypeFor(final ModuleItem<?> item) {
+
+        // FIXME why do I need this check in case of scripts?... Why should you
+        // not?
+        if (item.getWidgetStyle() != null
+                && item.getWidgetStyle().equals(StyleHook.COLUMNSELECTION)) {
```

Comment:
One easy way to avoid this issue is to write:
```java
if (StyleHook.COLUMNSELECTION.equals(item.getWidgetStyle())) {
```

And then you don't need the nesting or explicit null test.